mobile-app/app/components/SummaryTitle.tsx

```diff
@@ -99,29 +99,53 @@ export function SummaryTitle(props: ISummaryTitleProps): JSX.Element {
             >
               {translate("screens/common", "From")}
             </ThemedTextV2>
-            <ThemedViewV2
-              dark={tailwind("bg-mono-dark-v2-200")}
-              light={tailwind("bg-mono-light-v2-200")}
-              style={tailwind(
-                "rounded-full pl-1 pr-2.5 py-1 flex flex-row items-center overflow-hidden ml-2",
-              )}
-            >
-              <RandomAvatar name={props.fromAddress} size={20} />
-              <ThemedTextV2
-                ellipsizeMode="middle"
-                numberOfLines={1}
-                style={[
-                  tailwind("text-sm font-normal-v2 ml-1"),
-                  {
-                    minWidth: 10,
-                    maxWidth: 108,
-                  },
-                ]}
-                testID="wallet_address"
+
+            {props.isEvmToken ? (
+              // EVM from address
+              <AddressEvmTag
+                customStyle="flex flex-row items-center rounded-full pr-2.5 py-1 ml-2 overflow-hidden"
+                testID="from_address_label"
               >
-                {props.fromAddressLabel ?? props.fromAddress}
-              </ThemedTextV2>
-            </ThemedViewV2>
+                <>
+                  <RandomAvatar name={props.fromAddress} size={20} />
+                  <ThemedTextV2
+                    ellipsizeMode="middle"
+                    numberOfLines={1}
+                    style={[
+                      tailwind("text-sm font-normal-v2 ml-1"),
+                      // eslint-disable-next-line react-native/no-inline-styles
+                      { minWidth: 10, maxWidth: 108 },
+                    ]}
+                    testID="wallet_address"
+                  >
+                    {props.fromAddressLabel ?? props.fromAddress}
+                  </ThemedTextV2>
+                </>
```

Comment:
can we create small component to avoid repetition ?

Reply:
Updated :)